backend/corpora/dataset_processing/process.py

```diff
@@ -251,27 +290,69 @@ def update_db(dataset_id, metadata: dict = None, processing_status: dict = None)
             processing_status_updater(session, dataset.processing_status.id, processing_status)
 
 
-def download_from_dropbox_url(dataset_uuid: str, dropbox_url: str, local_path: str) -> str:
-    """Given a dropbox url, download it to local_path.
+def download_from_source_uri(dataset_uuid: str, source_uri: str, local_path: str) -> str:
+    """Given a source URI, download it to local_path.
     Handles fixing the url so it downloads directly.
     """
 
-    file_url = from_url(dropbox_url)
+    file_url = from_url(source_uri)
     if not file_url:
-        raise ValueError(f"Malformed Dropbox URL: {dropbox_url}")
+        raise ValueError(f"Malformed source URI: {source_uri}")
+
+    # This is a bit ugly and should be done polymorphically instead, but Dropbox support will be dropped soon
```

Comment:
Dropbox support will be dropped in the near future, so figured it wasn't worth the effort.